lib/configuration.js

```diff
@@ -167,6 +167,15 @@ var Configuration = function(givenConfig, logger) {
    */
   this._version = version;
   /**
+   * Boolean flag indicating whether or not the configuration instance was able
+   * to find a set of usable credentials to attempt authorization against the
+   * Stackdriver API.
+   * @memberof Configuration
+   * @private
+   * @type {Boolean}
+   */
+  this._lacksValidCredentials = false;
```

Comment:
Is it possible for this value to be observed before a call to `_checkAuthConfiguration`? Since this defaults to `false` errors will think valid credentials are available if they have not yet been checked. Maybe it would be better to default to `true`.